tests/migrations/test_operations.py

```diff
@@ -3652,47 +3652,47 @@ def test_add_constraint_percent_escaping(self):
-        # "%" generated in startswith lookup should be escaped in a way that is
-        # considered a leading wildcard.
-        check = models.Q(name__startswith="Albert")
-        constraint = models.CheckConstraint(check=check, name="name_constraint")
-        operation = migrations.AddConstraint("Author", constraint)
-        to_state = from_state.clone()
-        operation.state_forwards(app_label, to_state)
-        with connection.schema_editor() as editor:
-            operation.database_forwards(app_label, editor, from_state, to_state)
-        Author = to_state.apps.get_model(app_label, "Author")
-        with self.assertRaises(IntegrityError), transaction.atomic():
-            Author.objects.create(name="Artur")
-        # Literal "%" should be escaped in a way that is not a considered a
-        # wildcard.
-        check = models.Q(rebate__endswith="%")
-        constraint = models.CheckConstraint(check=check, name="rebate_constraint")
-        operation = migrations.AddConstraint("Author", constraint)
-        from_state = to_state
-        to_state = from_state.clone()
-        operation.state_forwards(app_label, to_state)
-        Author = to_state.apps.get_model(app_label, "Author")
-        with connection.schema_editor() as editor:
-            operation.database_forwards(app_label, editor, from_state, to_state)
-        Author = to_state.apps.get_model(app_label, "Author")
-        with self.assertRaises(IntegrityError), transaction.atomic():
-            Author.objects.create(name="Albert", rebate="10$")
-        author = Author.objects.create(name="Albert", rebate="10%")
-        self.assertEqual(Author.objects.get(), author)
-        # Right-hand-side baked "%" literals should not be used for parameters
-        # interpolation.
-        check = ~models.Q(surname__startswith=models.F("name"))
-        constraint = models.CheckConstraint(check=check, name="name_constraint_rhs")
-        operation = migrations.AddConstraint("Author", constraint)
-        from_state = to_state
-        to_state = from_state.clone()
-        operation.state_forwards(app_label, to_state)
-        with connection.schema_editor() as editor:
-            operation.database_forwards(app_label, editor, from_state, to_state)
-        Author = to_state.apps.get_model(app_label, "Author")
-        with self.assertRaises(IntegrityError), transaction.atomic():
-            Author.objects.create(name="Albert", surname="Alberto")
+        checks = [
+            # "%" generated in startswith lookup should be escaped in a way
+            # that is considered a leading wildcard.
+            (
+                models.Q(name__startswith="Albert"),
+                {"name": "Alberta"},
+                {"name": "Artur"},
+            ),
+            # Literal "%" should be escaped in a way that is not a considered a
+            # wildcard.
+            (models.Q(rebate__endswith="%"), {"rebate": "10%"}, {"rebate": "10%$"}),
```

Comment:
`10$` use to fail because the badly escaped constraint `%%\%%` still prevented it while `10%$` is only invalid if the constraint is properly created with `LIKE '%\%`.